qgis/examples/elevation_tile_downloader/core/task.py

```diff
@@ -1,117 +1,122 @@
-from qgis.core import (QgsTask,
-                       QgsMessageLog,
-                       Qgis,
-                       QgsRasterLayer,
-                       QgsProject,
-                       QgsNetworkAccessManager,
-                       QgsNetworkReplyContent,
-                       QgsGeometry
-                       )
-from PyQt5.QtCore import QFileInfo, QUrl, pyqtSignal
-from qgis.PyQt.QtNetwork import QNetworkRequest
-from typing import List
+from typing import List, Optional, Set
 from pathlib import Path
-import traceback
 import gzip
 from io import BytesIO
 
+from qgis.gui import QgisInterface
+from qgis.core import (
+    QgsTask,
+    QgsMessageLog,
+    Qgis,
+    QgsRasterLayer,
+    QgsProject,
+    QgsNetworkAccessManager,
+    QgsNetworkReplyContent,
+    QgsGeometry,
+)
+from qgis.utils import iface
+from qgis.PyQt.QtCore import QUrl
+from qgis.PyQt.QtWidgets import QMessageBox
+from qgis.PyQt.QtNetwork import QNetworkRequest
+
+iface: QgisInterface
 
-class DownloadTask(QgsTask):
-    next_tile = pyqtSignal(int)
```

Comment:
this signal is not really necessary IMO.. also not so sure if that doesn't mess with the main thread somehow, emitting signals from a running task. probably not though, the task sends out signals too, hm, just a thought..

Reply:
Yeah this was thought of as additional info for the user when the `setProgress` informed about the current tile only. But now that is switched to set the progress each time the next tile download starts this is redundant. Emitting signals from tasks and connecting to them in the main thread should be fine though

Reply:
> Emitting signals from tasks and connecting to them in the main thread should be fine though

I think so too